crates/swc_common/src/syntax_pos/hygiene.rs

```diff
@@ -289,19 +177,13 @@ impl Default for HygieneData {
 impl HygieneData {
     pub(crate) fn new() -> Self {
         HygieneData {
```

Comment:
Because neither `MarkData.parent` or `MarkData.builtin`  is used through the codebase. Also, `data.marks` is only changed in `Mark::fresh` before, let's think about what is `Mark::fresh` did, it get "parent" `Mark` , and put it in `data.marks` arrray and use latest `data.marks.len() - 1` to create global unique `Mark`, if the data of `data.marks` is never used, we could just use a `u32` to do the same thing, every time we just increase one and create a new mark

Reply:
I search the whole codebase the `parent param` of `Mark::fresh` actually does not mean any special thing, It always pass `Mark::root()`, It is just a placeholder IMO. Using `u32` instead of `Vec` will reduce a lot of lock time because a register operation is much cheaper than a memory operation.

Reply:
I know it's not used in main swc codebase at now, but it was used and it will be used